yt/fields/local_fields.py

```diff
@@ -1,50 +1,47 @@
-import warnings
-
+from yt.funcs import iterable
 from yt.utilities.logger import ytLogger as mylog
 
 from .field_info_container import FieldInfoContainer
 from .field_plugin_registry import register_field_plugin
 
 
 class LocalFieldInfoContainer(FieldInfoContainer):
-    def add_field(self, name, function=None, sampling_type=None, **kwargs):
-        if not isinstance(name, tuple):
-            if kwargs.setdefault("particle_type", False):
-                name = ("all", name)
+    def add_field(self, name, function, sampling_type, **kwargs):
+
+        sampling_type = self._sanitize_sampling_type(
+            sampling_type, kwargs.get("particle_type")
+        )
+
+        if isinstance(name, str) or not iterable(name):
+            if sampling_type == "particle":
+                ftype = "all"
             else:
-                name = ("gas", name)
+                ftype = "gas"
+            name = (ftype, name)
+
         override = kwargs.get("force_override", False)
         # Handle the case where the field has already been added.
         if not override and name in self:
-            mylog.error(
-                "Field %s already exists. To override use " + "force_override=True.",
-                name,
+            mylog.warning(
+                "Field %s already exists. To override use force_override=True.", name,
```

Comment:
Just to be consistent with the above-mentioned suggestion
```suggestion
               "Field %s already exists. To override use `force_override=True`.",
```